### PR TITLE
Improve singleton

### PR DESCRIPTION
**Pull request by SinghRajenM**
Kind of followup for commits fc9dfc86fc620d45edd546d9b75710452b241f61, 3dbb2c4b8ef47fddfd4e78aed086b28108fd036d, 3643a1875b588e730d132a3e401f099a83b866ef

### CODE REVIEW

File: PowerEditor/src/Parameters.h [hunk 1]
```diff
@@ -1648,6 +1648,14 @@ class NppParameters final
 	NppParameters();
```
Comment (donho) [inline PowerEditor/src/Parameters.h]: `NppParameters() = default;` ?

File: PowerEditor/src/MISC/Common/Common.h [hunk 2]
```diff
@@ -114,14 +114,18 @@ class WcharMbcsConvertor final
...
 protected:
-	WcharMbcsConvertor() {}
-	~WcharMbcsConvertor() {}
+	WcharMbcsConvertor() = default;
+	~WcharMbcsConvertor() = default;
 
 	// Since there's no public ctor, we need to void the default assignment operator and copy ctor.
 	// Since these are marked as deleted does not matter under which access specifier are kept
 	WcharMbcsConvertor(const WcharMbcsConvertor&) = delete;
 	WcharMbcsConvertor& operator= (const WcharMbcsConvertor&) = delete;
```
Comment (donho) [inline PowerEditor/src/MISC/Common/Common.h]: "No copy ctor and assignment" missing ?

File: PowerEditor/src/ScitillaComponent/Buffer.h [hunk 3]
```diff
@@ -122,7 +122,18 @@ class FileManager final
 		int _encoding;
 		EolType _eolFormat;
 	};
+
+	FileManager() = default;
 	~FileManager();
```
Comment (donho) [inline PowerEditor/src/ScitillaComponent/Buffer.h]: `~FileManager() = default;` ?

File: PowerEditor/src/Parameters.h (hunk 1 above)
Comment (SinghRajenM) [reply PowerEditor/src/Parameters.h]: Both of them can't be default as these already have user defined ctor and dtor.

File: PowerEditor/src/ScitillaComponent/Buffer.h (hunk 3 above)
Comment (SinghRajenM) [reply PowerEditor/src/ScitillaComponent/Buffer.h]: FileManager dtor can't be made default as user defined dtor already present which de-allocate a vector of `Buffer`.

**Comment by donho** [comment]
@SinghRajenM  Any comment ?

**Comment by SinghRajenM** [comment]
> 
> 
> @SinghRajenM Any comment ?

@donho it seems you missed notification as I didn't tag you in comments and also I have resolved the conversation.

Now I have marked both comments as "unresolved". Please check and let me know if further clarification if needed.

**Pull request closed without merging**